genai-perf/genai_perf/inputs/converters/rankings_converter.py

```diff
@@ -60,27 +65,25 @@ def convert(self, generic_dataset: GenericDataset, config: InputsConfig) -> Dict
 
             passage_entry = passages_data.rows[query_index]
 
+            passages: Union[List[str], List[Dict[str, str]]]
+            payload: Dict[str, Any]
```

Comment:
Just curious, why list the types here? Can't someone reading the code just jump to a definition to find that? That way python stays pythonic/simple?

Reply:
Mypy complains when statically checking types, because the if/else branches have different types for the same variables. This was the fix. I could use different variable names for passages (though I think that's more confusing), but payload will raise an error either way since the dictionary values are different types.